code/espurna/garland.cpp

```diff
@@ -0,0 +1,172 @@
+    root["garlandVisible"] = 1;
+}
+
+//------------------------------------------------------------------------------
+bool _garlandWebSocketOnKeyCheck(const char* key, JsonVariant& value) {
+    if (strncmp(key, NAME_GARLAND_ENABLED, strlen(NAME_GARLAND_ENABLED)) == 0) return true;
+    if (strncmp(key, NAME_GARLAND_BRIGHTNESS, strlen(NAME_GARLAND_BRIGHTNESS)) == 0) return true;
+    return false;
+}
+
+//------------------------------------------------------------------------------
+void _garlandWebSocketOnAction(uint32_t client_id, const char* action, JsonObject& data) {
+    if (strcmp(action, NAME_GARLAND_SWITCH) == 0) {
+        if (data.containsKey("status") && data.is<int>("status")) {
+            const char* payload = data["status"];
+            if (strlen(payload) == 1) {
+                if (payload[0] == '0') {
+                    _garland_enabled = false;
+                    pixels.clear();
+                    pixels.show();
+                } else if (payload[0] == '1')
+                    _garland_enabled = true;
+            }
+        }
+    }
+
+    if (strcmp(action, NAME_GARLAND_SET_BRIGHTNESS) == 0) {
+        if (data.containsKey("brightness")) {
+            byte new_brightness = data.get<byte>("brightness");
+            DEBUG_MSG_P(PSTR("[GARLAND] new brightness = %d\n"), new_brightness);
+            scene.setBrightness(new_brightness);
+        }
+    }
+}
+#endif
+
+//------------------------------------------------------------------------------
+// Loop
+//------------------------------------------------------------------------------
+void garlandLoop(void) {
+    if (!garlandEnabled())
+        return;
+
+    if (scene.run()) {
+    }
+
+    if (millis() - _last_update > _interval_effect_update) {
+        _last_update = millis();
+        _interval_effect_update = random(EFFECT_UPDATE_INTERVAL_MIN, EFFECT_UPDATE_INTERVAL_MAX);
+
+        int prevAnimInd = animInd;
+        while (prevAnimInd == animInd) animInd = random(ANIMS);
+
+        scene.setAnim(anims[animInd]);
+
+        int prevPalInd = paletteInd;
+        while (prevPalInd == paletteInd) paletteInd = random(PALS);
+        scene.setPalette(pals[paletteInd]);
+
+        DEBUG_MSG_P(PSTR("[GARLAND] Anim: %-10s Pal: %d Inter: %d avg_calc: %d avg_show: %d\n"), anims[animInd]->getName().c_str(), paletteInd, _interval_effect_update, scene.getAvgCalcTime(), scene.getAvgShowTime());
+        scene.setup();
+    }
+}
+
+//------------------------------------------------------------------------------
+void garlandSetup() {
+    garlandConfigure();
+
+// Websockets
+#if WEB_SUPPORT
+    wsRegister()
+        .onConnected(_garlandWebSocketOnConnected)
+        .onKeyCheck(_garlandWebSocketOnKeyCheck)
+        .onAction(_garlandWebSocketOnAction);
+#endif
+
+    espurnaRegisterLoop(garlandLoop);
+    espurnaRegisterReload(_garlandReload);
+
+    pixels.begin();
+    randomSeed(analogRead(0) * analogRead(1));
```

Comment:
Not valid on 8266, there's only one analog pin. There is `secureRandom` though, which ignores the seed